src/wallet/rpcnames.cpp

```diff
@@ -286,6 +289,71 @@ name_list ()
   );
 }
 
+/**
+ * Generate a salt using HKDF for a given name + private key combination.
+ * Indirectly used in name_new and name_firstupdate.
+ * Refactored out to make testing easier.
+ * @param key Private key.
+ * @param name The name.
+ * @param rand Set to the salt.
+ * @return True on success, false otherwise
+ */
+bool
+getNameSalt(const CKey& key, const valtype name, valtype& rand)
+{
+    const valtype ikm(key.begin(), key.end());
+    const std::string salt(reinterpret_cast<const char*>(name.data()), name.size());
+    const std::string info("Namecoin Registration Salt");
+    CHKDF_HMAC_SHA256_L32 hkdf32(ikm.data(), ikm.size(), salt);
+    unsigned char tmp[32];
+    hkdf32.Expand32(info, tmp);
+
+    rand = valtype(tmp, tmp + 20);
+    return true;
+}
+
+namespace
+{
+
+/**
+ * Generate a salt using HKDF for a given name + txout combination.
+ * Used in name_new and name_firstupdate.
+ * @param key Private key.
```

Comment:
Nit:  There is no `key` argument here.  I think it is not needed to explicitly list all `@param`, since they are self-explanatory anyway; so just removing them is probably the simplest fix.

Reply:
Removed in both functions